drop-storage/src/lib.rs

```diff
@@ -1172,4 +1213,103 @@ mod tests {
 
         assert_eq!(transfers.len(), 0);
     }
+
+    #[test]
+    fn remove_outgoing_rejected_file() {
+        let logger = slog::Logger::root(slog::Discard, slog::o!());
+        let storage = Storage::new(logger, ":memory:").unwrap();
+
+        let transfer_id: Uuid = "23e488a4-0521-11ee-be56-0242ac120002".parse().unwrap();
+
+        let transfer = TransferInfo {
+            id: transfer_id,
+            peer: "5.6.7.8".to_string(),
+            files: TransferFiles::Outgoing(vec![
+                TransferOutgoingPath {
+                    file_id: "id3".to_string(),
+                    size: 1024,
+                    base_path: "/dir".to_string(),
+                    relative_path: "3".to_string(),
+                },
+                TransferOutgoingPath {
+                    file_id: "id4".to_string(),
+                    relative_path: "4".to_string(),
+                    base_path: "/dir".to_string(),
+                    size: 2048,
+                },
+            ]),
+        };
+
+        storage.insert_transfer(&transfer).unwrap();
+        storage
+            .insert_outgoing_path_reject_state(transfer_id, "id3", false)
+            .unwrap();
+        assert!(storage
+            .remove_transfer_file(transfer_id, "id3")
+            .unwrap()
+            .is_some());
+        assert!(storage
+            .remove_transfer_file(transfer_id, "id4")
+            .unwrap()
+            .is_none());
+
+        let transfers = storage.transfers_since(0).unwrap();
+        assert_eq!(transfers.len(), 1);
+
+        let paths = match &transfers[0].transfer_type {
+            DbTransferType::Outgoing(out) => out,
+            _ => panic!("Unexpected transfer type"),
+        };
+        assert_eq!(paths.len(), 1); // 1 since we removed one of them
+        assert_eq!(paths[0].file_id, "id4");
```

Comment:
minor: it's worth adding an `assert_eq!` before removal and after to be sure it was really added and then removed :)

Reply:
Done